curator-recipes/src/main/java/org/apache/curator/framework/recipes/leader/LeaderSelector.java

```diff
@@ -341,11 +342,41 @@ public Participant getLeader() throws Exception
 
     static Participant getLeader(CuratorFramework client, Collection<String> participantNodes) throws Exception
     {
+        Participant result = null;
+        
         if ( participantNodes.size() > 0 )
         {
-            return participantForPath(client, participantNodes.iterator().next(), true);
+            Iterator<String> iter = participantNodes.iterator();
+            while ( iter.hasNext() )
+            {
+                
+                try
+                {
+                    result = participantForPath(client, iter.next(), true);
```

Comment:
Nah, looks good.  What I was talking about totally depends on which style you find more readable.  I'm not from the school of single-exit; to me multiple return points is easier to follow than mentally tracking variable state.  I would probably have written this:

```java
static Participant getLeader(CuratorFramework client, Collection<String> participantNodes) throws Exception {
    for (String node : participantNodes) {
        Participant result = participantForPath(client, node, true);
        if ( result != null )  {
            return result;
        }
    }
    return new Participant();
}
```

But what you got works just as well.